### PR TITLE
Detail screen

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 75413: Detail Screen](https://edu-redmine.sun-asterisk.vn/issues/75413)
## WHAT
- Detail Screen
## Evidence (Screenshot or Video)

https://github.com/awesome-academy/HCM_Android_2024_Weather_App_PJ2/assets/159242052/95845635-8bca-48d8-ae6c-50dc1f3b6ea5

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/blob/master/eng/android/coding_convention_android_kotlin.md | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(Other notes)*

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: app/src/main/java/com/example/weather/ui/adapter/DailyAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import com.example.weather.data.model.entity.WeatherBasic
+import com.example.weather.databinding.LayoutItemDailyBinding
+import com.example.weather.utils.Utils.getIcon
+import com.example.weather.utils.ext.kelvinToCelsius
+import com.example.weather.utils.ext.unixTimestampToDateMonthString
+import com.example.weather.utils.ext.unixTimestampToHourString
+
+class DailyAdapter : RecyclerView.Adapter<DailyAdapter.ViewHolder>() {
+
+    private val mListWeathers by lazy { mutableListOf<WeatherBasic>() }
```
Comment (daolq-2712) [inline app/src/main/java/com/example/weather/ui/adapter/DailyAdapter.kt]:
```suggestion
    private val mListWeathers by lazy { mutableListOf<WeatherBasic>() }
```
mListWeathers -> typo

sửa thành ``mWeathers`` hoặc ``mListWeather``

File: app/src/main/java/com/example/weather/ui/adapter/HourlyAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import com.example.weather.utils.Utils.getIcon
+import com.example.weather.utils.ext.kelvinToCelsius
+import com.example.weather.utils.ext.unixTimestampToHourString
+import com.example.weather.utils.ext.unixTimestampToTimeString
+import com.example.weather.utils.listener.OnItemClickListener
+
+class HourlyAdapter(var listener: OnItemClickListener, val context: Context) :
+    RecyclerView.Adapter<HourlyAdapter.ViewHolder>() {
+
+    private val mListWeathers by lazy { mutableListOf<WeatherBasic>() }
```
Comment (daolq-2712) [inline app/src/main/java/com/example/weather/ui/adapter/HourlyAdapter.kt]: .

File: app/src/main/java/com/example/weather/ui/adapter/HourlyAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    private var mPrePosition = -1
+    var mPairedList: MutableList<Pair<WeatherBasic, Int>> = mutableListOf<Pair<WeatherBasic, Int>>().apply {
```
Comment (daolq-2712) [inline app/src/main/java/com/example/weather/ui/adapter/HourlyAdapter.kt]:
```suggestion
    private var mPrePosition = -1
    
    var mPairedList: MutableList<Pair<WeatherBasic, Int>> = mutableListOf<Pair<WeatherBasic, Int>>().apply {
```

File: app/src/main/java/com/example/weather/ui/detail/DetailFragment.kt [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+    private val hourlyAdapter: HourlyAdapter by lazy {
+        HourlyAdapter(this, requireContext())
+    }
+    private val dailyAdapter: DailyAdapter by lazy {
+        DailyAdapter()
+    }
+    private var mHourlyPosition = 0
+
+    private lateinit var mWeather: Weather
```
Comment (daolq-2712) [inline app/src/main/java/com/example/weather/ui/detail/DetailFragment.kt]:
```suggestion
    private var mHourlyPosition = 0
    private lateinit var mWeather: Weather
```

File: app/src/main/java/com/example/weather/ui/detail/DetailFragment.kt [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+    override fun initData() {
+        arguments?.let { args ->
+            mWeather = args.getParcelable(Constant.WEATHER_KEY)!!
+            bindDataToView(mWeather)
+        }
+    }
+
+    override fun bindData() {
+        // todo implement later
```
Comment (daolq-2712) [inline app/src/main/java/com/example/weather/ui/detail/DetailFragment.kt]:
```suggestion
        // TODO implement later
```

File: app/src/main/res/layout/layout_daily.xml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:textColor="@color/white"
+        android:src="@drawable/ic_calendar"
+        android:layout_marginTop="@dimen/dp_5"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintTop_toTopOf="parent"/>
+
+    <View
+        android:id="@+id/viewSeparate"
```
Comment (daolq-2712) [inline app/src/main/res/layout/layout_daily.xml]:
```suggestion
        android:id="@+id/view_separate"
```

File: app/src/main/res/layout/layout_daily.xml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        android:layout_width="@dimen/dp_0"
+        android:layout_height="@dimen/dp_25"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/next_forecast" />
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/recyclerViewDaily"
```
Comment (daolq-2712) [inline app/src/main/res/layout/layout_daily.xml]:
```suggestion
        android:id="@+id/recyclerView_daily"
```

Comment (daolq-2712) [reply app/src/main/res/layout/layout_daily.xml]: thông nhất 1 style đặt id trong xml

File: app/src/main/res/layout/layout_footer_detail.xml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+            android:layout_width="@dimen/dp_20"
+            android:layout_height="@dimen/dp_20"
+            android:src="@drawable/ic_sun"
+            android:layout_marginStart="@dimen/dp_10"
+            android:layout_marginEnd="@dimen/dp_10"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintEnd_toStartOf="@id/hi_weather"/>
+
+        <TextView
+            android:id="@+id/hi_weather"
```
Comment (daolq-2712) [inline app/src/main/res/layout/layout_footer_detail.xml]:
```suggestion
            android:id="@+id/tv_weather"
```

**Review by daolq-2712 — CHANGES_REQUESTED** [review]

**Review by daolq-2712 — APPROVED** [review]
OK!

**Pull request merged**